pkg/network/server_test.go

```diff
@@ -1081,11 +1081,13 @@ func TestVerifyNotaryRequest(t *testing.T) {
 func TestTryInitStateSync(t *testing.T) {
 	t.Run("module inactive", func(t *testing.T) {
 		s := startTestServer(t)
+		defer s.Shutdown()
```

Comment:
Seems that this problem is the same as https://github.com/nspcc-dev/neo-go/pull/3307#discussion_r1486101955, so we need to refactor all code related to `startWithCleanup` functionality, because currently it just doesn't work.